### PR TITLE
React setup

### PR DESCRIPTION
**Pull request by julesntare**
#### What does this PR do?
Setup react project files
#### Description of Task to be completed
 - The project is configured from scratch using Webpack 4 & Babel.
 - Set up React Router and created a dummy route "/" for the landing page.
 - Added a ReadMe file which includes: Installation and run commands
 - Configure ESLint
#### How should this be manually tested?
clone Github repository 
checkout to origin branch `react-setup`
run `npm i` to install all dependencies
run `npm run dev` to test development mode
run `npm run lint` to test linting errors
#### Any background context you want to provide?
N/A
#### What are the relevant stories?
https://trello.com/c/sGpi3IWu/32-setup-react-project
#### Screenshot
N/A

#### Labels
ready to be reviewed

### CODE REVIEW

**Comment by julesntare** [comment]
[Setup React project](https://trello.com/c/sGpi3IWu/32-setup-react-project)

**Review by maximeish — APPROVED** [review]
great job jules. I'll have to modify my scripts I wrote when setting up testing according to the ones you used

**Comment by julesntare** [comment]
> great job jules. I'll have to modify my scripts I wrote when setting up testing according to the ones you used

I didn't add test script

**Comment by maximeish** [comment]
> > great job jules. I'll have to modify my scripts I wrote when setting up testing according to the ones you used
> 
> I didn't add test script

I mean like changing the script for running dev mode. Or let me delete them they will be replaced with those

**Comment by julesntare** [comment]
> > > great job jules. I'll have to modify my scripts I wrote when setting up testing according to the ones you used
> > 
> > 
> > I didn't add test script
> 
> I mean like changing the script for running dev mode. Or let me delete them they will be replaced with those

No problem I'll resolve if there comes conflicts

**Review by IRANKUND — APPROVED** [review]
great job bro

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	extends: ['eslint:recommended', 'plugin:react/recommended'],
+	parserOptions: {
+		ecmaFeatures: {
+			jsx: true,
+		},
+		ecmaVersion: 12,
+		sourceType: 'module',
+	},
+	plugins: ['react'],
+	rules: {},
```
Comment (malaba6) [inline .eslintrc.js]: Define rules here. For example the indent rule,...

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	extends: ['eslint:recommended', 'plugin:react/recommended'],
```
Comment (malaba6) [inline .eslintrc.js]: Remove this line, it's the same as line 7

File: src/index.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+</head>
+
+<body>
+    <noscript> You need to enable JavaScript to run this app. </noscript>
+    <!-- your react app will render here -->
+    <div id="app"></div>
+</body>
+
+</html>
```
Comment (malaba6) [inline src/index.html:17]: anable eslint to avoid such errors

**Review by malaba6 — CHANGES_REQUESTED** [review]

**Comment by malaba6** [comment]
Also please, squash the commits into one commit. We should have only one commit per PR to make it easier for those who will have to rebase from your work.

**Review by IRANKUND — APPROVED** [review]

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+const path = require("path");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+
+module.exports = {
```
Comment (maximeish) [inline webpack.config.js:6]: I think this looks fine, I am going to modify this file in ch-docker branch by adding
`devServer: {
 port: 8080,
 host: localhost
}`

to recognise the running port for docker

**Review by maximeish — APPROVED** [review]

File: src/components/Nav.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+=======
+        <ul>
+          <li>
+            <NavLink to="/">Home</NavLink>
+          </li>
+          <li>
+            <NavLink to="/about">About</NavLink>
+          </li>
+        </ul>
+>>>>>>> a56d48e... eslint fixed, and navbar added css, change directory of app.css
```
Comment (maximeish) [inline src/components/Nav.js:30]: Can you resolve this conflict. It's the one causing problems when we run tests

Comment (maximeish) [inline src/components/Nav.js:30]: @joshwambere Can you resolve this conflict. It's the one causing problem when we run tests

Comment (joshwambere) [reply src/components/Nav.js:30]: it has been resolved on bg-reactSetup-dep branch  except it hasn't been merged yet

Comment (maximeish) [reply src/components/Nav.js:30]: Ok no problem then. everything should work fine now

**Pull request merged**